### ISSUE TITLE
Not valid HTML markup?

### ISSUE THREAD

**Issue by alvarotrigo** [mbest/knockout.punches#53]
It looks quite nice! I like the approach!
Although it seems that using this plugin the HTML markup won't validate anymore when using things such as:

``` html
<ul>
    {{#foreach: items}}
        <li>The current item is: {{name}}</li>
    {{/foreach}}
</ul>
```

I guess the approach to it is by combining it with knockout binds?

``` html
<ul>
    <!-- foreach: items -->
        <li>The current item is: {{name}}</li>
    <!-- /ko -->
</ul>
```

**Comment by mbest** [comment]
You are right. In those cases, you can use the comment syntax or just include a data binding in the `<ul>`:

```
<ul data-bind="foreach: items">
    <li>The current item is: {{name}}</li>
</ul>
```

**Comment by mbest** [comment]
Also see knockout/knockout#1806

**Comment by IanYates** [comment]
You should also note that if you try to use {{}} syntax in certain areas, like between <tr> tags, the browser will (unhelpfully) move the text outside of the table for you, ruining your markup.  The fix there is the same - use the comment syntax.

**Comment by crissdev** [comment]
> You should also note that if you try to use {{}} syntax in certain areas, like between tags, the browser will (unhelpfully) move the text outside of the table for you, ruining your markup. The fix there is the same - use the comment syntax.

This really should go into docs.

**Comment by jlaustill** [comment]
So I actually prefer the standard ko data-bind="foreach: items" syntax verses {{#foreach: items}} {{/foreach}}.  I am really only interested in using text and html attributes like the first two examples on the documentation.  Both of those seem to produce valid markup for me.  Is it possible to load them separately?  They both seem to be included with ko.punches.attributeInterpolationMarkup.enable()?  

The attributes doesn't create valid markup either, which is fine, I'm not sold on that being better than the attr method.

I haven't played with the dynamic namespaced bindings yet, but I don't see any reason that they wouldn't create valid markup? 

I agree with crissdev, this should go in the docs.  It would be great to see a section that clearly lays out what does/doesn't create valid markup.  That being said, if the ONLY thing this provided me was text and html attributes, I would be happy with this addition.  That alone cleans my code up an AMAZING amount.
